common/src/main/java/com/wynntils/wc/utils/scoreboard/objectives/ObjectiveManager.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.utils.scoreboard.objectives;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.event.WynntilsScoreboardUpdateEvent;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.server.ServerScoreboard;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ObjectiveManager {
+    // §b is guild objective, §a is normal objective and §c is daily objective
+    public static final Pattern OBJECTIVE_PATTERN = Pattern.compile("^§([abc])[- ]\\s§7(.*): *§f(\\d+)§7/(\\d+)$");
+
+    private static WynnObjective guildWynnObjective = null;
+
+    private static final List<WynnObjective> WYNN_OBJECTIVES = new ArrayList<>();
+
+    @SubscribeEvent
+    public static void onScoreboardUpdate(WynntilsScoreboardUpdateEvent event) {
+        if (!event.getChangeMap().containsKey(WynntilsScoreboardUpdateEvent.ChangeType.Objective)) return;
+
+        Set<WynntilsScoreboardUpdateEvent.Change> changes =
```

Comment:
Just thinking out loud here.. Do you think it would be better to have specialized Events instead, so you'd send like a `WynntilsScoreboardObjectiveUpdateEvent` instead, and don't have to filter out, and don't have to just get the relevant part of the map the first thing you do? It feels a bit like this is multiple different events baked into one.